testng-core/src/main/java/org/testng/internal/invokers/TestInvoker.java

```diff
@@ -660,29 +661,45 @@ private ITestResult invokeMethod(
               ? (IHookable) arguments.getInstance()
               : m_configuration.getHookable();
 
+      boolean willfullyIgnored = false;
+      boolean usesHookableInstance = hookableInstance != null;
       if (MethodHelper.calculateTimeOut(arguments.getTestMethod()) <= 0) {
-        if (hookableInstance != null) {
-          MethodInvocationHelper.invokeHookable(
-              arguments.getInstance(),
-              arguments.getParameterValues(),
-              hookableInstance,
-              thisMethod,
-              testResult);
+        if (usesHookableInstance) {
+          willfullyIgnored =
+              !MethodInvocationHelper.invokeHookable(
+                  arguments.getInstance(),
+                  arguments.getParameterValues(),
+                  hookableInstance,
+                  thisMethod,
+                  testResult);
         } else {
           // Not a IHookable, invoke directly
           MethodInvocationHelper.invokeMethod(
               thisMethod, arguments.getInstance(), arguments.getParameterValues());
         }
-        setTestStatus(testResult, ITestResult.SUCCESS);
+        boolean testStatusRemainedUnchanged = testResult.isNotRunning();
+        if (usesHookableInstance && willfullyIgnored && testStatusRemainedUnchanged) {
+          throw new TestNotInvokedException(arguments.tm);
+        } else {
+          setTestStatus(testResult, ITestResult.SUCCESS);
+        }
       } else {
         // Method with a timeout
-        MethodInvocationHelper.invokeWithTimeout(
-            arguments.getTestMethod(),
-            arguments.getInstance(),
-            arguments.getParameterValues(),
-            testResult,
-            hookableInstance);
+        willfullyIgnored =
+            !MethodInvocationHelper.invokeWithTimeout(
+                arguments.getTestMethod(),
+                arguments.getInstance(),
+                arguments.getParameterValues(),
+                testResult,
+                hookableInstance);
+      }
+      boolean testStatusRemainedUnchanged = testResult.isNotRunning();
+      if (usesHookableInstance && willfullyIgnored && testStatusRemainedUnchanged) {
+        throw new TestNotInvokedException(arguments.tm);
```

Comment:
Why not directly set status and throwable here?

Reply:
Fixed